src/Command/InfectionCommand.php

```diff
@@ -203,26 +209,24 @@ protected function configure(): void
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $container = $this->getContainer();
-
-        if (!$container->get('coverage.checker')->hasDebuggerOrCoverageOption()) {
+        if (!$this->container['coverage.checker']->hasDebuggerOrCoverageOption()) {
             $this->consoleOutput->logMissedDebuggerOrCoverageOption();
 
             return 1;
         }
 
         /** @var InfectionConfig $config */
-        $config = $container->get('infection.config');
+        $config = $this->container['infection.config'];
```

Comment:
What is the benefit of this change? It's a step back from [Continainer PSR](https://www.php-fig.org/psr/psr-11/) compatibility.

Here and in other places below.

Reply:
Ultimately the goal would be typehinted getters from an immutable container as per [this comment](https://github.com/infection/infection/pull/718#issuecomment-506549634).

Reply:
Still don't get how [Container Interface](https://github.com/php-fig/container/blob/014d250daebff39eba15ba990eeb2a140798e77c/src/ContainerInterface.php) conflicts with immutable container, when it's only has `get` and `has` methods.

So, for the referenced comment, we can do:

```php
public function getInfectionConfig(): InfectionConfig
{
    return $this->container->get(InfectionConfig::class);
}
```

while with this PR it is:

```php
public function getInfectionConfig(): InfectionConfig
{
    return $this->container[InfectionConfig::class];
}
```

Reply:
>Still don't get how Container Interface conflicts with immutable container, when it only has get and has methods.

It doesn't, but it doesn't have typehints.

I'm confused by your example however.

What I did here is change the container we are using in the application. I.e. instead of consuming a generic PSR11 container in `InfectionCommand` and potential other commands, we use ours `InfectionContainer`, because instead of having:

```php
/** @var Foo $foo */
$foo = $container->get(Foo::class);
// Not typhint safe since there is no guarantee `$foo` is an 
// instance of `Foo` or that the docblock above is correct
```

We will have:

```php
$foo = $container->getFoo(); // return type `Foo`, so type safe
```

As of now, `InfectionContainer` _extends_ `Pimple\Container`. Since the later is mutable, I would like to move away from it. This however is for later and I'm not 100% on the specifics yet